### PR TITLE
Check if rules used in authSchema even exists

### PR DESCRIPTION
**Pull request by kerimcharfi**
Here's a quick and dirty POC for #73

### CODE REVIEW

File: packages/core/src/config.ts [hunk 1]
```diff
@@ -13,6 +13,41 @@ export interface IAuthZConfig {
...
   config: IAuthZConfig
 ): Omit<Required<IAuthZConfig>, 'authSchema'> & { authSchema?: AuthSchema } {
+
+  if (config.authSchema) {
+    // Check if rules in authSchema actually exist in rules definition
+    for (const typeName in config.authSchema) {
+      const typeRuleDefinitions = config.authSchema[typeName];
+      if ('__authz' in typeRuleDefinitions) {
+        // check type level rules
+        //@ts-ignore
```
Comment (XBeg9) [inline packages/core/src/config.ts:23]: try to not use ignore

File: packages/core/src/config.ts [hunk 2]
```diff
@@ -13,6 +13,41 @@ export interface IAuthZConfig {
...
+        let ruleName: string
+        for (ruleName of ruleNames as string[]) {
+          if (!(ruleName in config.rules)) {
+            throw new Error(`Rule ${ruleName} not found in rules`);
+          }
+        }
+      } else {
+        // check field level rules
+        for (const fieldName in typeRuleDefinitions) {
+          //@ts-ignore
```
Comment (XBeg9) [inline packages/core/src/config.ts:38]: try to not use ignore

File: packages/core/src/config.ts [hunk 3]
```diff
@@ -13,6 +13,41 @@ export interface IAuthZConfig {
 export function completeConfig(
   config: IAuthZConfig
 ): Omit<Required<IAuthZConfig>, 'authSchema'> & { authSchema?: AuthSchema } {
+
+  if (config.authSchema) {
```
Comment (XBeg9) [inline packages/core/src/config.ts:17]: can you add tests?

**Review by XBeg9 — CHANGES_REQUESTED** [review]

**Comment by XBeg9** [comment]
please add changesets

File: packages/core/src/config.ts (hunk 3 above)
Comment (kerimcharfi) [reply packages/core/src/config.ts:26]: sure

**Comment by kerimcharfi** [comment]
Alright?

**Review by XBeg9 — APPROVED** [review]

**Pull request merged**